source/loader/layers/sanitizer/linux/backtrace.cpp

```diff
@@ -25,8 +67,24 @@ StackTrace GetCurrentBacktrace() {
 
     StackTrace Stack;
     for (int i = 0; i < FrameCount; i++) {
-        BacktraceInfo addr_info(Symbols[i]);
-        Stack.stack.emplace_back(std::move(addr_info));
+        if (SymbolizeCode != nullptr) {
+            std::string Result;
+            std::string ModuleName = ExtractModuleName(Symbols[i]);
+            if (SymbolizeCode(ModuleName, (uint64_t)Frames[i], Result)) {
+                SourceInfo SrcInfo = ParseSymbolizerOutput(Result);
+                std::ostringstream OS;
+                if (SrcInfo.file != "??") {
+                    OS << "in " << SrcInfo.function << " " << SrcInfo.file
+                       << ":" << SrcInfo.line << ":" << SrcInfo.column;
+                } else {
+                    OS << "in " << SrcInfo.function << " (" << ModuleName << "+"
+                       << Frames[i] << ")";
+                }
+                Stack.stack.emplace_back(OS.str());
+                continue;
+            }
```

Comment:
I know I was a bit hasty in implementing this part at first, in fact, the symbolization should be invoked as late as possible.
Can you save as few as possible information when doing backtrace, and symbolize them when really need it?
Thank you!

Reply:
I think we can get handle of the module, not file path

Reply:
> I know I was a bit hasty in implementing this part at first, in fact, the symbolization should be invoked as late as possible. Can you save as few as possible information when doing backtrace, and symbolize them when really need it? Thank you!

I agree, the symbolization takes time, so we should only do it when necessary (in error reporting stage I think).

Reply:
nice suggestions. updated.